### PR TITLE
test(runtime): cover http network edge cases

### PR DESCRIPTION
**Pull request by danielraffel**
## Summary
- Adds a 39-assertion runtime HTTP/network coverage batch in `test/test_network_stream.cpp`.
- Covers invalid explicit ports, default/unsupported scheme behavior, refused loopback POST failures, `HttpResponse::ok()` boundaries, and `http_download` false paths.
- Based on latest `origin/main` at `f96a8745f2c1f76f7e7501cbe645fce8b77e3e8f` after #2793.

## Coverage context
- Current Codecov component evidence before this batch: `runtime` is below the 75% priority threshold.
- This PR is intentionally batched at 39 assertions to stay in the requested 36-48 fixes/assertions range and avoid extra CI runs.

## Validation
- `git diff --check origin/main...HEAD`
- `source tools/scripts/cli_version_check.sh && pulp_cli_version_check`
- `python3 tools/scripts/skill_sync_check.py --help`
- `python3 tools/scripts/version_bump_check.py --help`

Focused build/test and Codecov validation are delegated to Shipyard/GitHub runners per project workflow and current no-local-build constraint.

### CODE REVIEW

File: test/test_network_stream.cpp [hunk 1]
```diff
@@ -1133,6 +1133,122 @@ TEST_CASE("HttpStream factories and refetch reset closed state to request result
...
+    HttpServerRunner runner(server);
+    REQUIRE(runner.wait_until_running());
+    REQUIRE(wait_until_http_ready(port));
+
+    REQUIRE_FALSE(http_download("http://127.0.0.1:" + std::to_string(port) + "/not-found",
+                                (std::filesystem::temp_directory_path() /
+                                 "pulp-http-404-download.bin").string(),
+                                2));
+    REQUIRE_FALSE(std::filesystem::exists(std::filesystem::temp_directory_path() /
+                                          "pulp-http-404-download.bin"));
```
[the remark this answers is not in the document]
Comment (danielraffel) [reply test/test_network_stream.cpp]: Addressed in `41e169676`: the 404 negative case now uses a per-port temp path and removes stale state before asserting non-existence.
